src/passes/OptimizeInstructions.cpp

```diff
@@ -1345,15 +1377,57 @@ struct OptimizeInstructions
         }
       }
       // operations on all 1s
-      // TODO: shortcut method to create an all-ones?
       if (right->value == Literal(int32_t(-1)) ||
           right->value == Literal(int64_t(-1))) {
         if (binary->op == Abstract::getBinary(type, Abstract::And)) {
+          // x & -1   ==>   x
           return binary->left;
         } else if (binary->op == Abstract::getBinary(type, Abstract::Or) &&
                    !EffectAnalyzer(getPassOptions(), features, binary->left)
                       .hasSideEffects()) {
+          // x & -1   ==>   -1
           return binary->right;
+        } else if (binary->op == Abstract::getBinary(type, Abstract::Sub)) {
+          // x - (-1)   ==>   x + 1
+          Builder builder(*getModule());
+          return builder.makeBinary(
+            Abstract::getBinary(type, Abstract::Add),
+            binary->left,
+            builder.makeConst(Literal::makeFromInt32(1, type)));
```

Comment:
this can reuse nodes. we can change the sub's opcode to an add, and change the constant's value to 1.